test/e2e/networking.go

```diff
@@ -87,53 +96,51 @@ var _ = Describe("Networking", func() {
 			}
 		}()
 
-		By("Creating a replication controller")
-		rc, err := c.ReplicationControllers(ns).Create(&api.ReplicationController{
-			ObjectMeta: api.ObjectMeta{
-				Name: name,
-				Labels: map[string]string{
-					"name": name,
-				},
-			},
-			Spec: api.ReplicationControllerSpec{
-				Replicas: 8,
-				Selector: map[string]string{
-					"name": name,
-				},
-				Template: &api.PodTemplateSpec{
-					ObjectMeta: api.ObjectMeta{
-						Labels: map[string]string{"name": name},
+		By("Creating a webserver pod on each node")
+		for node := 0; node < peers; node++ {
+			Logf("Creating pod %s-%d on node %s", name, node, nodes.Items[node].Name)
+			_, err := c.Pods(ns).Create(&api.Pod{
+				ObjectMeta: api.ObjectMeta{
+					Name: fmt.Sprintf("%s-%d", name, node),
+					Labels: map[string]string{
+						"name": name,
 					},
-					Spec: api.PodSpec{
-						Containers: []api.Container{
-							{
-								Name:    "webserver",
-								Image:   "kubernetes/nettest:1.1",
-								Command: []string{"-service=" + name, "-namespace=" + ns},
-								Ports:   []api.ContainerPort{{ContainerPort: 8080}},
-							},
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  "webserver",
+							Image: "kubernetes/nettest:1.1",
+							Command: []string{
+								"-service=" + name,
+								fmt.Sprintf("-peers=%d", peers),
+								"-namespace=" + ns},
+							Ports: []api.ContainerPort{{ContainerPort: 8080}},
 						},
 					},
+					Host:          nodes.Items[node].Name,
+					RestartPolicy: api.RestartPolicyNever,
 				},
-			},
-		})
-		if err != nil {
-			Fail(fmt.Sprintf("unable to create test rc: %v", err))
+			})
+			Expect(err).NotTo(HaveOccurred())
 		}
-		// Clean up rc
+		// Clean up the pods
 		defer func() {
 			defer GinkgoRecover()
-			By("Cleaning up the replication controller")
-			// Resize the replication controller to zero to get rid of pods.
-			rcReaper, err := kubectl.ReaperFor("ReplicationController", c)
-			if err != nil {
-				Fail(fmt.Sprintf("unable to stop rc %v: %v", rc.Name, err))
-			}
-			if _, err = rcReaper.Stop(ns, rc.Name); err != nil {
-				Fail(fmt.Sprintf("unable to stop rc %v: %v", rc.Name, err))
+			By("Cleaning up the webserver pods")
+			for node := 0; node < peers; node++ {
+				if err = c.Pods(ns).Delete(fmt.Sprintf("%s-%d", name, node)); err != nil {
+					Logf("Failed to delete pod %s-%d: %v", name, node, err)
+				}
 			}
 		}()
 
+		By("Wait for the webserver pods to be ready")
+		for node := 0; node < peers; node++ {
+			err = waitForPodRunningInNamespace(c, fmt.Sprintf("%s-%d", name, node), ns)
```

Comment:
Yes, I know, and that's exactly what I do in the end to end test for cluster level logging. I decided to do something different here because I know how to compute the name from the index. However, I am happy to add a aux variable and store them. It just felt strange to store an array like ["wombat-0", "wombat-1", "wombat-2"].